bin/runner.js

```diff
@@ -19,75 +20,92 @@ var init = function(args){
   );
   var config = forFiles(verifyExamplesConfig, args._);
 
-  return {
-    config: config,
-    run: generate,
-    cleanup: cleanup,
-    runElmTest: runElmTest,
-    args: args,
+  return Object.assign(config, args, {
     testsDocPath: path.join(args.output, "VerifyExamples")
-  };
+  });
 };
 
 
 function generate(model, allTestsGenerated) {
-  if (model.args.warn) console.log('Generate tests from examples...');
-  var config = model.config;
+  if (model.warn) console.warn('Generate tests from examples...');
   cleanup(model);
 
-  if (config.tests.length === 0){
-    if (model.args.warn) {
-      console.log('No tests listed! Modify your elm-verify-examples.json file to include modules');
+  if (model.tests.length === 0){
+    if (model.warn) {
+      warn('No tests listed! Modify your elm-verify-examples.json file to include modules');
+      if (model.failOnWarn) process.exit(1);
     }
     return;
   }
 
-  var app = Elm.VerifyExamples.worker(config);
+  var app = Elm.VerifyExamples.worker(model);
 
-  app.ports.readFile.subscribe(function(test) {
+  app.ports.readFile.subscribe(function(moduleName) {
     var pathToModule = path.join(
-      config.testsPath,
-      config.root,
-      elmModuleToPath(test)
+      model.testsPath,
+      model.root,
+      elmModuleToPath(moduleName)
     );
     fs.readFile(
         pathToModule,
         "utf8",
-        function(err, data) {
+        function(err, fileText) {
       if (err) {
         console.error(err);
         process.exit(-1);
         return;
       }
-      app.ports.generateModuleVerifyExamples.send([test, data]);
+        app.ports.generateModuleVerifyExamples.send(
+          { moduleName: moduleName,
+            fileText: fileText,
+            ignoredWarnings: ignoredWarnings(model.ignoreWarnings, moduleName)
+          }
+        );
+    });
+  });
+
+  var warnings = [];
+  app.ports.warn.subscribe(function(args) {
+    var moduleName = args[0];
+    var warningsForModule = args[1];
```

Comment:
🎨 why not send out `Json.Encode.object [ ("moduleName", Json.Encode.string moduleName), ("warnings", Json.Encode.list <| List.map Json.Encode.string warnings)]` instead?